File: tests/refs/two_sum.py
"""Reference: two-sum returning the pair of indices summing to `target`.

The returned indices are in ascending order; returns [-1, -1] when no pair
exists.
"""

from __future__ import annotations
from typing import Dict, List


def twoSum(nums: List[int], target: int) -> List[int]:
    seen: Dict[int, int] = {}
    for i, x in enumerate(nums):
        complement = target - x
        if complement in seen:
            return [seen[complement], i]
        seen[x] = i
    return [-1, -1]

File: tests/refs/TwoSum.java
/*
 * Reference Java implementation for tests/cases/two_sum.json.
 * Returns ascending indices summing to target, or [-1, -1] if none.
 */
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static long[] twoSum(long[] nums, long target) {
        Map<Long, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            long complement = target - nums[i];
            Integer j = seen.get(complement);
            if (j != null) return new long[]{j, i};
            seen.put(nums[i], i);
        }
        return new long[]{-1, -1};
    }
}
